samtranslator/model/__init__.py

```diff
@@ -536,6 +536,25 @@ def _construct_tag_list(
         # customer's knowledge.
         return get_tag_list(sam_tag) + get_tag_list(additional_tags) + get_tag_list(tags)
 
+    def propagate_tags(
+        self, propagate_tags: Optional[bool], resources: List[Resource], tags: Optional[Dict[str, Any]]
+    ) -> None:
+        """
+        Propagates tags to the resources.
+
+        :param resources: List of resources
+        :param tags: dictionary of tags to propagate to the resources.
+
+        :return: None
+        """
+        if not propagate_tags:
+            return
+
+        tags_list = get_tag_list(tags)
+        for resource in resources:
```

Comment:
What if some of them cannot have Tags?

Reply:
That is checked on line 555. It checks if resource has `Tags` property.

Reply:
Actually, withdrawing this comment, Chris has much more specific feedback